### PR TITLE
Add bloqs for binary negation and twos complement negation

### PR DESCRIPTION
**Pull request by fpapa250**

### CODE REVIEW

File: qualtran/bloqs/arithmetic/multiplication.py [hunk 1]
```diff
@@ -544,3 +549,70 @@ def _square_real_number() -> SquareRealNumber:
     import_line='from qualtran.bloqs.arithmetic.multiplication import SquareRealNumber',
     examples=(_square_real_number,),
 )
+
+
+@frozen
+class Negate(Bloq):
+    """Flips all the bits of a register.
+    Args:
```
Comment (mpharrigan) [inline qualtran/bloqs/arithmetic/multiplication.py:557]: newline

File: qualtran/bloqs/arithmetic/multiplication.py [hunk 2]
```diff
@@ -544,3 +549,70 @@ def _square_real_number() -> SquareRealNumber:
     import_line='from qualtran.bloqs.arithmetic.multiplication import SquareRealNumber',
     examples=(_square_real_number,),
 )
+
+
+@frozen
+class Negate(Bloq):
+    """Flips all the bits of a register.
```
Comment (mpharrigan) [inline qualtran/bloqs/arithmetic/multiplication.py:556]: docstring should relate "flip all the bits" doc with "negate" name somewhere

File: qualtran/bloqs/arithmetic/multiplication.py [hunk 3]
```diff
@@ -544,3 +549,70 @@ def _square_real_number() -> SquareRealNumber:
...
+        return {'x': x}
+
+    def short_name(self) -> str:
+        return '~x'
+
+
+@frozen
+class NegateTwosComplement(Bloq):
+    """Negates an integer in two's complement form.
+        |x> => |-1 * x>
```
Comment (mpharrigan) [inline qualtran/bloqs/arithmetic/multiplication.py:587]: newline after first line of docstring

**Comment by anurudhp** [comment]
Closing this PR as these operations are now implemented, see #1144 (Negate), #1161 (BitwiseNot).

**Pull request closed without merging**